control/statesp.py

```diff
@@ -91,38 +91,21 @@ class StateSpace(LTI):
 
     """
 
-    def __init__(self, *args):
+    def __init__(self,A,B,C,D,dt=None):
         """Construct a state space object.
 
         The default constructor is StateSpace(A, B, C, D), where A, B, C, D are
         matrices or equivalent objects.  To call the copy constructor, call
         StateSpace(sys), where sys is a StateSpace object.
+        
+        Matrix sizes need to be consistent:
+        A must be square.
+        A and B must have the same number of rows.
+        A and C must have the same number of columns.
+        B and D must have the same number of columns.
+        C and D must have the same number of rows.
 
         """
-
-        if len(args) == 4:
-            # The user provided A, B, C, and D matrices.
-            (A, B, C, D) = args
-            dt = None;
-        elif len(args) == 5:
-            # Discrete time system
-            (A, B, C, D, dt) = args
-        elif len(args) == 1:
-            # Use the copy constructor.
-            if not isinstance(args[0], StateSpace):
```

Comment:
I suggest using `if isinstance(A, StateSpace)` to preserve this behavior for backwards compatibility.

Reply:
The point of the changes in the pull request is to remove the ambiguity of being able to define statespaces in two ways:

```
S1=StateSpace(A,B,C,D)
S2=StateSpace(oldstatespace)
```

Yes that will break code for people who use the second method. 

That probably should have been in the initial comment.

If this is breaking things too much for now, maybe it would be an option to create a "next major release" branch to include this change. That might make the transition obvious and it would make it easier to use the old version in the future.

If you don't want the module to be backwards incompatible at all, reject the pull request or tell me and I'll delete it.

Also your suggested change wouldn't work because the ```StateSpace.__init__ ```
```
__init__(A,B,C,D,dt=None):
```
would expect ```B,C,D``` and throw an error if you only gave it a StateSpace instance as first argument.

The only way to preserve the functionality of being able to define a StateSpace with both, a previous instance or a list of variables, is to keep the current code.